### ISSUE TITLE
Add session.interactive to easily determine if Nox is being run by a human.

### ISSUE THREAD

**Issue by theacodes** [theacodes/nox#189]
**How would this feature be useful?**
In many cases its useful to know if nox is being run in an actual terminal or via CI or some other "headless" means.

**Describe the solution you'd like**
A new property on `Session` called `interactive`.  It would look like this:

```python
@nox.session
def docs(session):
    if session.interactive:
        session.run("sphinx-autobuild", ...)
    else:
        session.run("sphinx")
```

**Comment by dhermes** [comment]
Why can't `nox` users just call `sys.stdout.isatty()`?

**Comment by theacodes** [comment]
They can. But why make it harder than it needs to be?

**Comment by theacodes** [comment]
Also, we could add flags such as ``--non-interactive`` to force `session.interactive` to be false.

**Comment by dhermes** [comment]
> But why make it harder than it needs to be?

Well `session.interactive` and `sys.stdout.isatty()` are about the same number of characters.

The `--non-interactive` flag is another story though.

**Comment by theacodes** [comment]
True, but one requires (a) an import and (b) remembering what it is. I
actually filed this issue after forgetting what it was and having to look
it up for the 20th time.

**Comment by AGeekInside** [comment]
I took a stab at adding the interactive attribute. There is a pull request with the initial changes to add the interactive attribute. I have not added the --non-interactive flag yet.

**Comment by theacodes** [comment]
This was implemented int #196
